utils/trivialize-rules/trivialize-rules.js

```diff
@@ -181,8 +182,43 @@ files.fork().zipAll([ sources.fork(), rules ]).map(([name, source, ruleset]) =>
 
   domains = Array.from(domains);
 
+  function isStaticCookie(securecookie) {
+    if (securecookie.host === '.+' && securecookie.name === '.+') {
+      return true;
+    }
+
+    if (!securecookie.host.startsWith('^') || !securecookie.host.endsWith('$')) {
+      return false;
+    }
+
+    let localDomains = new Set();
+
+    try {
+      explodeRegExp(securecookie.host, domain => {
+        if (domain.startsWith('.')) {
+          domain = domain.slice(1);
```

Comment:
Is this sufficient munging to ensure it matches the target domains?
If not, at worst it will not match and result in a `false` result, which isn't terrible but not desirable.

Reply:
@Hainish this should be good enough. As mentioned in https://github.com/EFForg/https-everywhere/pull/16029#issuecomment-404150837, securecookie.host only match  cookie.domain if there is a rule covers cookie.domain. Given the `target` are trivial, cannot be anything other than `domain.example.com` and `.domain.example.com` for a securecookie rule to works. With condition (1) effective, securecookie.host should explode to either one of them. Otherwise, the securecookie will never be applied (dangling rules?) under the current implementation. 

With the above logic in mind, I have also add 4bbd731to remove some of these securecookie rules that will never be effective.

Reply:
IMHO, it would be great to have #15998 and #16000 merged before running this against the master.